In [2]:
import pandas as pd
import numpy as np

In [3]:
DATA_RAW_DIR = "../../data/raw/"

In [4]:
fci = pd.read_csv(DATA_RAW_DIR + "FCI.csv", sep=";", encoding="utf-8-sig")
coloris = pd.read_csv(DATA_RAW_DIR + "COLORIS.csv", sep=";", encoding="utf-8-sig")
aero = pd.read_csv(DATA_RAW_DIR + "AERO.csv", sep=";", encoding="utf-8-sig")
vio = pd.read_csv(DATA_RAW_DIR + "VIO.csv", sep=";", encoding="utf-8-sig")

In [5]:
display(fci.info())
display(fci.head())

<class 'pandas.DataFrame'>
RangeIndex: 317 entries, 0 to 316
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   ID Ticket GENERGY       280 non-null    str    
 1   ID Ticket OCEANE        280 non-null    str    
 2   Description             280 non-null    str    
 3   Acteur                  279 non-null    str    
 4   Paramétrage             280 non-null    str    
 5   Statut                  280 non-null    str    
 6   Priorité                280 non-null    str    
 7   Actions réalisés        276 non-null    str    
 8   Date d'ouverture        280 non-null    str    
 9   Date de résolution      280 non-null    str    
 10  Durée de résolution     280 non-null    str    
 11  ID Jira                 12 non-null     str    
 12  Date de livraison JIRA  0 non-null      float64
 13  Unnamed: 13             0 non-null      float64
dtypes: float64(2), str(12)
memory usage: 34.8 KB


None

,ID Ticket GENERGY,ID Ticket OCEANE,Description,Acteur,Paramétrage,Statut,Priorité,Actions réalisés,Date d'ouverture,Date de résolution,Durée de résolution,ID Jira,Date de livraison JIRA,Unnamed: 13
0,INC001010838599,2510J75674,Annulation commande dans suivi FCI,TERCHELLAH Rim,Support FCI,Résolu,P4,01/10_SN3 :Les commande ont bien été annulées,01/10/2025,01/10/2025,✅ Résolu en < 1 jour,NaN,NaN,NaN
1,INC001010842749,2510K34156,ETAT DIFFERENT ENTRE FCI ET COLORIS : COMMANDE...,TERCHELLAH Rim,Support FCI,Résolu,P4,02/10_SN3 : La commande à bien été mis à jour.,02/10/2025,02/10/2025,✅ Résolu en < 1 jour,NaN,NaN,NaN
2,INC001010843892,2510K68517,BdC HEPOC pour le client FREE SAS,TERCHELLAH Rim,Support FCI,Résolu,P3,06/10_SN3 :Le BdC HEPOC a bien été ajouté à FR...,03/10/2025,06/10/2025,✅ Résolu en 1 jour,NaN,NaN,NaN
3,INC001010844022,2510K75077,Demande de mise à jour d'état,TERCHELLAH Rim,Support FCI,Résolu,P4,03/10_SN3 : La commande à bien été mis à jour.,03/10/2025,03/10/2025,✅ Résolu en < 1 jour,NaN,NaN,NaN
4,INC001010844444,2510K77715,Fonction recherche NRA inoperante,TERCHELLAH Rim,Support FCI,Résolu,P3,06/10_SN3 : Le problème venait du Banco et il ...,03/10/2025,06/10/2025,✅ Résolu en 1 jour,NaN,NaN,NaN


In [6]:
display(coloris.info())
display(coloris.head())

<class 'pandas.DataFrame'>
RangeIndex: 309 entries, 0 to 308
Data columns (total 14 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   ID Ticket GENERGY     282 non-null    str  
 1   ID Ticket OCEANE      285 non-null    str  
 2   Description           285 non-null    str  
 3   Acteur                285 non-null    str  
 4   Colonne1              286 non-null    str  
 5   Statut                286 non-null    str  
 6   Priorité              286 non-null    str  
 7   Actions réalisés      280 non-null    str  
 8   Date d'ouverture      286 non-null    str  
 9   Date de résolution    286 non-null    str  
 10  Durée de résolution   286 non-null    str  
 11  ID Jira               20 non-null     str  
 12  Offre composée        80 non-null     str  
 13  Catégorie d'incident  113 non-null    str  
dtypes: str(14)
memory usage: 33.9 KB


None

,ID Ticket GENERGY,ID Ticket OCEANE,Description,Acteur,Colonne1,Statut,Priorité,Actions réalisés,Date d'ouverture,Date de résolution,Durée de résolution,ID Jira,Offre composée,Catégorie d'incident
0,INC001010841667,2510K27774,Blocage de FAS sur commandes,ZAKRAOUI Mohamed,Support COLORIS,Résolu,P4,02/10 SN3: Un point planifié avec le demandeur...,02/10/2025,06/10/2025,✅ Résolu en 2 jours,NaN,NaN,NaN
1,INC001010842399,2510K36669,Transfert Coloris-OPUS en rejet,ZAKRAOUI Mohamed,Support COLORIS,Résolu,P4,02/10 SN3:il s’agissait d’un problème lié à OJ...,02/10/2025,02/10/2025,✅ Résolu en < 1 jour,NaN,NaN,NaN
2,INC001010766880,2508V23293,Blocage de resil emplacement,PIERROT CALLIZO Celia,Paramétrage COLORIS,Résolu,P4,En cours de modification pour une mep prevue m...,25/08/2025,06/10/2025,✅ Résolu en 30 jours,NaN,NaN,NaN
3,INC001010785806,2509Z08065,Absence champ type de module dans proxima,PIERROT CALLIZO Celia,Paramétrage COLORIS,Transféré,P4,"Suite analyse, la modification du CI PROXIMA e...",04/09/2025,02/10/2025,➡️ Transféré en 20 jours,NaN,HBNRO,Transféré
4,INC001010835596,2510K20044,6 prestations production alors que commande d...,PIERROT CALLIZO Celia,Paramétrage COLORIS,Résolu,P4,NaN,02/10/2025,09/10/2025,✅ Résolu en 5 jours,NaN,NaN,Paramétrage COLORIS


In [7]:
display(aero.info())
display(aero.head())

<class 'pandas.DataFrame'>
RangeIndex: 193 entries, 0 to 192
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID Ticket GENERGY    54 non-null     str    
 1   TYPE                 54 non-null     str    
 2   ELEMENT              54 non-null     str    
 3   Description          54 non-null     str    
 4   Acteur               53 non-null     str    
 5   Catégorie            54 non-null     str    
 6   Statut               54 non-null     str    
 7   Priorité             54 non-null     str    
 8   Actions réalisés     54 non-null     str    
 9   Date d'ouverture     54 non-null     str    
 10  Date de résolution   54 non-null     str    
 11  Durée de résolution  54 non-null     float64
 12  ID Jira              13 non-null     str    
dtypes: float64(1), str(12)
memory usage: 19.7 KB


None

,ID Ticket GENERGY,TYPE,ELEMENT,Description,Acteur,Catégorie,Statut,Priorité,Actions réalisés,Date d'ouverture,Date de résolution,Durée de résolution,ID Jira
0,mail,AERO,FIBRE IHM-FTTE,Pb dysfonctionnement Eligibilité FTTE sur RIP ...,Akram Sahli,Demande d'assistance,Résolu,P3,"Après vérification, les résultats d’éligibilit...",01/12/2025 00:00,02/12/2025 00:00,2.0,NaN
1,INC001010977278,AERO,FIBRE IHM-FTTH,Il est indiqué que le code immeuble IMB/33119/...,Mariem Hammami,Demande d'assistance,Résolu,P3,l’IMB n’est pas présente dans les fichiers del...,19/12/2025,19/12/2025 00:00,1.0,NaN
2,INC001010986955,AERO,Orange-Eligibility KO,WS Orange KO deouis 14:50,Akram Sahli,Bug,Résolu,P3,Le problème a été résolu hier à 17h40.\nToutef...,29/12/2025,29/12/2025,1.0,NaN
3,INC001011008164,AERO,DONNEES ABSENTES OU INCORRECTES,Nous constatons une erreur sur le PM lorsqu'on...,Akram Sahli,Hors perimètre d'AERO,Résolu,P3,le PM OAPC n'est pas dans notre périmètre d'in...,14/01/2026,14/01/2026,1.0,NaN
4,INC001011016378,AERO,CONNEXION IMPOSSIBLE,Depuis 19/01/2026 08h00 on a erreurs 400 sur P...,Akram Sahli,Aucun probléme lié a AERO,Résolu,P3,"Après analyse des logs, les requêtes ne semble...",20/01/2026,21/01/2026 00:00,2.0,NaN


In [8]:
display(vio.info())
display(vio.head())

<class 'pandas.DataFrame'>
RangeIndex: 190 entries, 0 to 189
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype
---  ------               --------------  -----
 0   ID Ticket GENERGY    4 non-null      str  
 1   ID Ticket OCEANE     180 non-null    str  
 2   App                  180 non-null    str  
 3   Description          180 non-null    str  
 4   Acteur               180 non-null    str  
 5   Périmètre            180 non-null    str  
 6   Statut               180 non-null    str  
 7   Priorité             180 non-null    str  
 8   Actions réalisés     180 non-null    str  
 9   Date d'ouverture     180 non-null    str  
 10  Date de résolution   180 non-null    str  
 11  Durée de résolution  180 non-null    str  
 12  ID Jira              6 non-null      str  
dtypes: str(13)
memory usage: 19.4 KB


None

,ID Ticket GENERGY,ID Ticket OCEANE,App,Description,Acteur,Périmètre,Statut,Priorité,Actions réalisés,Date d'ouverture,Date de résolution,Durée de résolution,ID Jira
0,INC001010961694,2512K27406,Parc,Problème Marine,Mariem Hammami,Support VIO,Résolu,P3,L’investigation a été effectuée,10/11/2025,11/12/2025 00:00,✅ Résolu en 23 jours,NaN
1,INC001010946366,2512H51825,Parc,Dans le cadre d'une mise à jour du SI et du tr...,Akram Sahli,Support VIO,Résolu,P4,Nous avons bien pris en compte votre demande c...,02/12/2025,04/12/2025,✅ Résolu en 2 jours,NaN
2,INC001010954206,2512J75412,Parc,J'ai besoin du parc d'accès du client CEGETEL ...,Akram Sahli,Support VIO,Résolu,P3,Nous avons bien pris en compte votre demande c...,09/12/2025,10/12/2025,✅ Résolu en 1 jour,NaN
3,NaN,2512L11587,Vigie,"[VIGIE_VI] / [SMC4, BRONZE, SOX] / [REMCOLL_NO...",Akram Sahli,Support VIO,Résolu,P4,"Bonjour,\nL’incident a été rétabli le lendemai...",12/12/2025,18/12/2025 00:00,✅ Résolu en 4 jours,NaN
4,NaN,2512L01771,Sagic,"[SAGIC] / [SMC3, ARGENT] / [OBV_APP_FILE_STRIN...",Mariem Hammami,Support VIO,Résolu,P3,Voir avec la MOE si le problème est récurrent....,12/12/2025,19/01/1900,✅ Résolu en < 1 jour,NaN


## Shared helpers

Used by every per-application section below: French accented text normalization, title
truncation, the `actions_realisees` → `TransferDestination` matcher, and the `Offre composée`
offer/version parser.

In [9]:
import re
import sys
import unicodedata

sys.path.insert(0, "../..")
from app.modules.ticket_management.domain.enums.transfer_destination import TransferDestination

DEST_NAME_TO_VALUE = {member.name: member.value for member in TransferDestination}


def normalize_text(text: str) -> str:
    text = unicodedata.normalize("NFKD", text)
    text = "".join(ch for ch in text if not unicodedata.combining(ch))
    return text.lower()


TITLE_MAX_LEN = 80


def truncate_title(description: str) -> str:
    description = description.strip()
    if len(description) <= TITLE_MAX_LEN:
        return description
    return description[: TITLE_MAX_LEN - 1].rstrip() + "…"


_DUREE_RE = re.compile(r"(Résolu|Transféré) en (?:(< 1)|(\d+)) jours?")


def parse_duree_resolution(value: str) -> tuple[str, int]:
    """Parses the 'Durée de résolution' display column into (action_word, day_count)."""
    match = _DUREE_RE.search(value)
    if match is None:
        raise ValueError(f"Unrecognized 'Durée de résolution' format: {value!r}")
    action, lt_one, days = match.groups()
    return action, 0 if lt_one else int(days)


# Applications with a support/paramétrage team split, matched via "paramétrage <app>" /
# "support <app>" (in either word order).
TEAM_QUALIFIED_APPS = {
    "COLORIS": ("CONFIG_COLORIS", "SUPPORT_COLORIS"),
    "FCI": ("CONFIG_FCI", "SUPPORT_FCI"),
}

# Destinations matched as a bare keyword, using each enum member's own value. BANCO is handled
# separately below since it also shows up as an incidental root-cause mention.
SIMPLE_KEYWORD_DESTINATIONS = [
    "AERO", "VIO", "EEP", "CLIP", "ULYSSE", "ACACIA", "SANTAFE", "PROXIMA", "HABILITATION",
]


def _detect_destination(norm: str) -> tuple[str | None, str | None]:
    for app, (config_name, support_name) in TEAM_QUALIFIED_APPS.items():
        app_norm = normalize_text(app)
        if re.search(rf"parametrage.{{0,20}}{app_norm}|{app_norm}.{{0,20}}parametrage", norm):
            return config_name, None
        if re.search(rf"support.{{0,20}}{app_norm}", norm):
            return support_name, None
    for name in SIMPLE_KEYWORD_DESTINATIONS:
        value_norm = normalize_text(TransferDestination[name].value)
        if re.search(rf"\b{re.escape(value_norm)}\b", norm):
            return name, None
    if re.search(r"\bcatalogue\b", norm):
        return None, "catalogue_not_in_enum"
    # Bare app mention with no support/paramétrage qualifier maps to that app's support team.
    for app, (config_name, support_name) in TEAM_QUALIFIED_APPS.items():
        app_norm = normalize_text(app)
        if re.search(rf"\b{app_norm}\b", norm):
            return support_name, None
    return None, None


def match_transfer_destination(text: str | float) -> tuple[str | None, str | None]:
    """Matches actions_realisees free text to a TransferDestination enum member name.

    Returns (destination_name, flag_reason). flag_reason is set when the text mentions a
    destination that has no corresponding enum value, or mentions no destination at all.

    Some actions_realisees cells discuss more than one team/app before stating the actual
    destination (e.g. "...impact sur le paramétrage dans PROXIMA... Il n'y a pas d'incohérence
    dans le paramétrage COLORIS... Ticket transféré à Proxima."). To avoid matching an
    incidental earlier mention instead of the actual stated destination, destinations are first
    searched in a window right after each "transfér*" occurrence, most recent first (the last
    action log entry is normally the one that states the outcome); only if no such window yields
    a match does the search fall back to the whole text.
    """
    if not isinstance(text, str) or not text.strip():
        return None, "no_actions_text"
    norm = normalize_text(text)

    anchors = [m.start() for m in re.finditer(r"transf\w*", norm)]
    for start in reversed(anchors):
        dest, flag = _detect_destination(norm[start : start + 100])
        if dest is not None or flag is not None:
            return dest, flag

    # "Banco" is also a common root-cause mention in resolved tickets, so it only counts as a
    # destination here when it appears right after a "transfer(red)" word.
    if re.search(r"transfer\w*.{0,15}banco", norm):
        return "BANCO", None

    dest, flag = _detect_destination(norm)
    if dest is not None or flag is not None:
        return dest, flag

    # No destination stated in the text at all.
    return "DEVELOPMENT_TEAM", None


def parse_offer_version(raw: str | float) -> tuple[str | None, str | None]:
    """Parses the 'Offre composée' column into (offer, version).

    Multiple slash-separated offers keep only the first. A trailing "V<n>" token (possibly
    followed by more text, e.g. "V14 et V16") is taken as the version; everything before it is
    the offer. Offer names that don't match an Offer enum member, and version values that don't
    match a Version enum member, are still returned as-is - the enums are incomplete and are
    expected to be extended separately.
    """
    if not isinstance(raw, str) or not raw.strip():
        return None, None
    text = raw.strip()
    if "/" in text:
        text = text.split("/")[0].strip()
    match = re.search(r"\bV(\d+)\b", text, flags=re.IGNORECASE)
    if match:
        offer = text[: match.start()].strip()
        return (offer or None), f"V{match.group(1)}"
    return text, None

## FCI preprocessing

Raw export is semicolon-separated, UTF-8 with a BOM (`sep=";"`, `encoding="utf-8-sig"`).

`Date de livraison JIRA` and `Unnamed: 13` are 100% null (317/317) and are dropped as export artifacts.
Blank export rows (37 of them) are identified by a null `Statut` — every real ticket row has one,
so this is a more reliable filter than requiring both ID columns to be present (some real rows are
missing one of the two IDs but still have a status).

In [10]:
fci = fci.drop(columns=["Date de livraison JIRA", "Unnamed: 13"])
fci = fci.dropna(subset=["Statut"]).reset_index(drop=True)

fci = fci.rename(columns={
    "ID Ticket GENERGY ": "genergy_id",
    "ID Ticket OCEANE": "oceane_id",
    "Description": "description",
    "Acteur": "acteur",
    "Paramétrage": "team_raw",
    "Statut": "statut_raw",
    "Priorité": "priorite_raw",
    "Actions réalisés": "actions_realisees",
    "Date d'ouverture": "date_ouverture_raw",
    "Date de résolution": "date_resolution_raw",
    "Durée de résolution": "duree_resolution_raw",
    "ID Jira": "jira_id",
})

print(fci.shape)
fci.head()

(280, 12)

,genergy_id,oceane_id,description,acteur,team_raw,statut_raw,priorite_raw,actions_realisees,date_ouverture_raw,date_resolution_raw,duree_resolution_raw,jira_id
0,INC001010838599,2510J75674,Annulation commande dans suivi FCI,TERCHELLAH Rim,Support FCI,Résolu,P4,01/10_SN3 :Les commande ont bien été annulées,01/10/2025,01/10/2025,✅ Résolu en < 1 jour,NaN
1,INC001010842749,2510K34156,ETAT DIFFERENT ENTRE FCI ET COLORIS : COMMANDE...,TERCHELLAH Rim,Support FCI,Résolu,P4,02/10_SN3 : La commande à bien été mis à jour.,02/10/2025,02/10/2025,✅ Résolu en < 1 jour,NaN
2,INC001010843892,2510K68517,BdC HEPOC pour le client FREE SAS,TERCHELLAH Rim,Support FCI,Résolu,P3,06/10_SN3 :Le BdC HEPOC a bien été ajouté à FR...,03/10/2025,06/10/2025,✅ Résolu en 1 jour,NaN
3,INC001010844022,2510K75077,Demande de mise à jour d'état,TERCHELLAH Rim,Support FCI,Résolu,P4,03/10_SN3 : La commande à bien été mis à jour.,03/10/2025,03/10/2025,✅ Résolu en < 1 jour,NaN
4,INC001010844444,2510K77715,Fonction recherche NRA inoperante,TERCHELLAH Rim,Support FCI,Résolu,P3,06/10_SN3 : Le problème venait du Banco et il ...,03/10/2025,06/10/2025,✅ Résolu en 1 jour,NaN


In [11]:
# jira_id: strip stray trailing whitespace seen in raw values (e.g. "FCI-10066 ")
fci["jira_id"] = fci["jira_id"].str.strip()
fci["requires_jira"] = fci["jira_id"].notna()

fci[["jira_id", "requires_jira"]].value_counts(dropna=False)

jira_id    requires_jira
NaN        False            268
FCI-10128  True               3
FCI-10175  True               2
FCI-10066  True               1
FCI-10061  True               1
FCI-10067  True               1
FCI-10098  True               1
FCI-10197  True               1
FCI-10589  True               1
FCI-10655  True               1
Name: count, dtype: int64

In [12]:
# Dates are dd/mm/yyyy, with a "dd/mm/yyyy HH:MM" variant on ~40 rows (all exactly 00:00 in this
# file, i.e. no real time-of-day info) - format="mixed" handles both.
fci["date_ouverture"] = pd.to_datetime(fci["date_ouverture_raw"], format="mixed", dayfirst=True)
fci["date_resolution"] = pd.to_datetime(fci["date_resolution_raw"], format="mixed", dayfirst=True)

fci[["date_ouverture_raw", "date_ouverture", "date_resolution_raw", "date_resolution"]].head()

,date_ouverture_raw,date_ouverture,date_resolution_raw,date_resolution
0,01/10/2025,2025-10-01,01/10/2025,2025-10-01
1,02/10/2025,2025-10-02,02/10/2025,2025-10-02
2,03/10/2025,2025-10-03,06/10/2025,2025-10-06
3,03/10/2025,2025-10-03,03/10/2025,2025-10-03
4,03/10/2025,2025-10-03,06/10/2025,2025-10-06


In [13]:
# duree_resolution_raw is a derived/display column, not a Ticket field - parsed only to
# cross-check consistency against (date_resolution - date_ouverture).
parsed = fci["duree_resolution_raw"].apply(parse_duree_resolution)
fci["duree_action"] = parsed.apply(lambda t: t[0])
fci["duree_jours"] = parsed.apply(lambda t: t[1])

# Cross-check 1: action word in the duration string should agree with Statut
action_to_statut = {"Résolu": "Résolu", "Transféré": "Transféré"}
mismatch_statut = fci[fci["duree_action"].map(action_to_statut) != fci["statut_raw"]]
print("Statut / durée action mismatches:", len(mismatch_statut))

# Cross-check 2: parsed day count vs actual calendar day difference
fci["date_diff_jours"] = (fci["date_resolution"] - fci["date_ouverture"]).dt.days
mismatch_days = fci[fci["duree_jours"] != fci["date_diff_jours"]]
print("Duration / date-diff mismatches:", len(mismatch_days))
mismatch_days[["genergy_id", "date_ouverture", "date_resolution", "duree_resolution_raw", "date_diff_jours"]]

Statut / durée action mismatches: 0
Duration / date-diff mismatches: 81


,genergy_id,date_ouverture,date_resolution,duree_resolution_raw,date_diff_jours
2,INC001010843892,2025-10-03,2025-10-06,✅ Résolu en 1 jour,3
4,INC001010844444,2025-10-03,2025-10-06,✅ Résolu en 1 jour,3
5,INC001010847232,2025-10-07,2025-11-07,✅ Résolu en 23 jours,31
11,INC001010854963,2025-10-09,2025-10-13,✅ Résolu en 2 jours,4
34,INC001010855663,2025-10-10,2025-11-05,✅ Résolu en 18 jours,26
...,...,...,...,...,...
253,INC001011177741,2026-05-07,2026-06-04,✅ Résolu en 20 jours,28
259,INC001011214471,2026-06-05,2026-06-10,✅ Résolu en 3 jours,5
260,INC001011213770,2026-06-05,2026-06-08,✅ Résolu en 1 jour,3
261,INC001011213771,2026-05-11,2026-06-08,✅ Résolu en 20 jours,28


In [14]:
# Priorité values (P1-P4) already match Priority enum values exactly - direct passthrough.
assert set(fci["priorite_raw"].unique()) <= {"P1", "P2", "P3", "P4"}
fci["priority"] = fci["priorite_raw"]

# Statut -> Status enum. Raw data only ever has these two terminal states.
STATUT_MAP = {"Résolu": "RESOLVED", "Transféré": "TRANSFERRED"}
assert set(fci["statut_raw"].unique()) <= set(STATUT_MAP)
fci["status"] = fci["statut_raw"].map(STATUT_MAP)

# "Paramétrage" column actually encodes functional_team, not a free category:
# its only two values are "Support FCI" (-> SUPPORT) and "Paramétrage" (-> CONFIGURATION),
# matching the TransferDestination naming (SUPPORT_FCI / CONFIG_FCI).
TEAM_MAP = {"Support FCI": "SUPPORT", "Paramétrage": "CONFIGURATION"}
assert set(fci["team_raw"].unique()) <= set(TEAM_MAP)
fci["functional_team"] = fci["team_raw"].map(TEAM_MAP)

fci[["priorite_raw", "priority", "statut_raw", "status", "team_raw", "functional_team"]].drop_duplicates()

,priorite_raw,priority,statut_raw,status,team_raw,functional_team
0,P4,P4,Résolu,RESOLVED,Support FCI,SUPPORT
2,P3,P3,Résolu,RESOLVED,Support FCI,SUPPORT
8,P3,P3,Transféré,TRANSFERRED,Support FCI,SUPPORT
9,P3,P3,Transféré,TRANSFERRED,Paramétrage,CONFIGURATION
13,P2,P2,Résolu,RESOLVED,Support FCI,SUPPORT
25,P4,P4,Résolu,RESOLVED,Paramétrage,CONFIGURATION
29,P4,P4,Transféré,TRANSFERRED,Support FCI,SUPPORT
60,P3,P3,Résolu,RESOLVED,Paramétrage,CONFIGURATION
64,P2,P2,Transféré,TRANSFERRED,Support FCI,SUPPORT
91,P1,P1,Résolu,RESOLVED,Paramétrage,CONFIGURATION


### `duree_resolution_raw`

Derived/display column (emoji + action word + day count), e.g. "✅ Résolu en 23 jours" /
"➡️ Transféré en < 1 jour". Not a `Ticket` field — parsed only to cross-check against
`date_resolution - date_ouverture`. The two disagree on 81/280 rows; the parsed day count is
consistently lower on cross-week-boundary spans, indicating `duree_resolution_raw` counts business
days while the date columns are calendar days. Not stored downstream.

### Field mapping decisions

- `title` — no separate title field in the source; built by truncating `description`
  (`TITLE_MAX_LEN`), full text kept in `description`.
- `category` — no source column for it in the FCI export; fixed to `Category.CATEGORY_VIDE` for
  every row.
- `assignee_id` — `acteur` (raw person name) is kept as-is; mapping names to Auth `User` UUIDs is
  the seeding script's responsibility, not this notebook's, to keep this output free of any
  infra/DB dependency.
- `transferred_to` — parsed from `actions_realisees` free text (see below); no destination column
  exists in the source.
- `resolution_notes` — `actions_realisees` copied as one string; not split into `Comment` entities.
- Final `status` — every row is migrated as `CLOSED` regardless of original Résolu/Transféré;
  `original_status` retains the source distinction.
- `created_at`/`resolved_at`/`closed_at` — localized to `Europe/Paris`; the source date columns
  have no real time-of-day in this file (all midnight where a time is present at all).

In [15]:
fci["title"] = fci["description"].apply(truncate_title)

# category has no source column in the FCI export - fixed default for every row.
fci["category"] = "vide"

fci[["description", "title", "category"]].head()

,description,title,category
0,Annulation commande dans suivi FCI,Annulation commande dans suivi FCI,vide
1,ETAT DIFFERENT ENTRE FCI ET COLORIS : COMMANDE...,ETAT DIFFERENT ENTRE FCI ET COLORIS : COMMANDE...,vide
2,BdC HEPOC pour le client FREE SAS,BdC HEPOC pour le client FREE SAS,vide
3,Demande de mise à jour d'état,Demande de mise à jour d'état,vide
4,Fonction recherche NRA inoperante,Fonction recherche NRA inoperante,vide


### `transferred_to` matching rules

Destinations found in `actions_realisees` text across the 63 `Transféré` rows:
- "paramétrage/parametrage COLORIS" → `CONFIG_COLORIS`; "paramétrage/parametrage FCI" → `CONFIG_FCI`;
  "support COLORIS" → `SUPPORT_COLORIS`; "EEP" → `EEP`; "ACACIA" → `ACACIA`; "Habilitation" →
  `HABILITATION`; "transféré ... à BANCO" → `BANCO`.
- Bare "COLORIS" mentions with no support/paramétrage qualifier → `SUPPORT_COLORIS`.
- "Catalogue" / "équipe Catalogue" mentions have no corresponding `TransferDestination` value and
  are left flagged (`catalogue_not_in_enum`) rather than guessed.
- Rows where the text never states a destination (e.g. INC001010947598, discusses a Banco-related
  root cause with no "transferred to X" wording) are routed to `DEVELOPMENT_TEAM`.

In [16]:
transferred_mask = fci["status"] == "TRANSFERRED"
results = fci.loc[transferred_mask, "actions_realisees"].apply(match_transfer_destination)
fci["transferred_to"] = None
fci["transfer_destination_flag"] = None
fci.loc[transferred_mask, "transferred_to"] = results.apply(lambda t: t[0])
fci.loc[transferred_mask, "transfer_destination_flag"] = results.apply(lambda t: t[1])

print(fci.loc[transferred_mask, "transferred_to"].value_counts(dropna=False))
print()
print(fci.loc[transferred_mask, "transfer_destination_flag"].value_counts(dropna=False))

transferred_to
CONFIG_COLORIS      31
EEP                 12
NaN                  6
SUPPORT_COLORIS      5
DEVELOPMENT_TEAM     3
CONFIG_FCI           2
ACACIA               2
BANCO                1
HABILITATION         1
Name: count, dtype: int64

transfer_destination_flag
NaN                      57
catalogue_not_in_enum     6
Name: count, dtype: int64


### Final status and timestamps

Every migrated row is closed regardless of the original Résolu/Transféré distinction:
`status = CLOSED`, `closed_at = date_resolution`. `original_status` retains RESOLVED/TRANSFERRED,
and `resolved_at`/`transferred_to` are populated only for rows that actually went through that path
(matching what `Ticket.resolve()`/`Ticket.transfer()` would set before `close()`).

`resolution_notes` is `actions_realisees` copied as-is (applies to both resolved and transferred
rows — it's a general action log, not resolution-specific text).

In [17]:
fci["original_status"] = fci["status"]  # RESOLVED / TRANSFERRED, kept for traceability
fci["status"] = "CLOSED"

fci["created_at"] = fci["date_ouverture"].dt.tz_localize("Europe/Paris")
fci["closed_at"] = fci["date_resolution"].dt.tz_localize("Europe/Paris")
fci["resolved_at"] = fci["closed_at"].where(fci["original_status"] == "RESOLVED")
fci["updated_at"] = fci["closed_at"]

fci["resolution_notes"] = fci["actions_realisees"]

fci[[
    "genergy_id", "original_status", "status", "created_at", "resolved_at",
    "closed_at", "transferred_to",
]].head(10)

,genergy_id,original_status,status,created_at,resolved_at,closed_at,transferred_to
0,INC001010838599,RESOLVED,CLOSED,2025-10-01 00:00:00+02:00,2025-10-01 00:00:00+02:00,2025-10-01 00:00:00+02:00,None
1,INC001010842749,RESOLVED,CLOSED,2025-10-02 00:00:00+02:00,2025-10-02 00:00:00+02:00,2025-10-02 00:00:00+02:00,None
2,INC001010843892,RESOLVED,CLOSED,2025-10-03 00:00:00+02:00,2025-10-06 00:00:00+02:00,2025-10-06 00:00:00+02:00,None
3,INC001010844022,RESOLVED,CLOSED,2025-10-03 00:00:00+02:00,2025-10-03 00:00:00+02:00,2025-10-03 00:00:00+02:00,None
4,INC001010844444,RESOLVED,CLOSED,2025-10-03 00:00:00+02:00,2025-10-06 00:00:00+02:00,2025-10-06 00:00:00+02:00,None
5,INC001010847232,RESOLVED,CLOSED,2025-10-07 00:00:00+02:00,2025-11-07 00:00:00+01:00,2025-11-07 00:00:00+01:00,None
6,INC001010852029,RESOLVED,CLOSED,2025-10-08 00:00:00+02:00,2025-10-09 00:00:00+02:00,2025-10-09 00:00:00+02:00,None
7,INC001010857716,RESOLVED,CLOSED,2025-10-10 00:00:00+02:00,2025-10-10 00:00:00+02:00,2025-10-10 00:00:00+02:00,None
8,INC001010850889,TRANSFERRED,CLOSED,2025-10-10 00:00:00+02:00,NaT,2025-10-10 00:00:00+02:00,CONFIG_COLORIS
9,INC001010856055,TRANSFERRED,CLOSED,2025-10-10 00:00:00+02:00,NaT,2025-10-10 00:00:00+02:00,NaN


### Enum value consistency

All enum-shaped columns are written using the domain enum's *value* string (e.g. `priority = "P4"`
= `Priority.LOW.value`, `category = "vide"` = `Category.CATEGORY_VIDE.value`, `status = "CLOSED"` =
`Status.CLOSED.value`). `transferred_to` was written as the enum *member name* by the matcher above,
so it's mapped to its value here (`DEST_NAME_TO_VALUE`, defined in Shared helpers) for the same
convention.

In [18]:
fci["transferred_to"] = fci["transferred_to"].map(DEST_NAME_TO_VALUE)

fci.loc[fci["original_status"] == "TRANSFERRED", ["genergy_id", "transferred_to", "transfer_destination_flag"]]

,genergy_id,transferred_to,transfer_destination_flag
8,INC001010850889,Paramétrage COLORIS,NaN
9,INC001010856055,NaN,catalogue_not_in_enum
15,INC001010866834,NaN,catalogue_not_in_enum
16,INC001010867102,Paramétrage COLORIS,NaN
23,INC001010880501,NaN,catalogue_not_in_enum
...,...,...,...
271,INC001011227267,EEP,NaN
272,INC001011228913,EEP,NaN
277,INC001011231373,EEP,NaN
278,INC001011234838,EEP,NaN


## Final assembly and export

Assembles the processed columns into a shape close to the `Ticket` entity.

Fields that aren't a 1:1 mapping to `Ticket`:
- `acteur` is the raw name string, not `assignee_id` — UUID mapping happens in the seeding script.
- `original_status` and `transfer_destination_flag` are diagnostic columns, not `Ticket` fields —
  they preserve the source Résolu/Transféré distinction and mark rows the seeding script needs to
  handle specially (`catalogue_not_in_enum`).
- `application` is fixed to `"FCI"`; `offer`/`version`/`element`/`vio_app` are `None` (not
  applicable outside COLORIS/AERO/VIO); `operational_highlight` defaults to `False` (no source
  signal); `archived_at` is `None` (nothing in this export represents archival).

In [19]:
fci_processed = pd.DataFrame({
    "genergy_id": fci["genergy_id"],
    "oceane_id": fci["oceane_id"],
    "title": fci["title"],
    "description": fci["description"],
    "application": "FCI",
    "status": fci["status"],
    "priority": fci["priority"],
    "category": fci["category"],
    "functional_team": fci["functional_team"],
    "acteur": fci["acteur"],
    "created_at": fci["created_at"],
    "updated_at": fci["updated_at"],
    "resolved_at": fci["resolved_at"],
    "closed_at": fci["closed_at"],
    "resolution_notes": fci["resolution_notes"],
    "transferred_to": fci["transferred_to"],
    "jira_id": fci["jira_id"],
    "requires_jira": fci["requires_jira"],
    "jira_delivery_date": None,
    "operational_highlight": False,
    "offer": None,
    "version": None,
    "element": None,
    "vio_app": None,
    "archived_at": None,
    # diagnostics for the seeding script / manual review - not Ticket fields
    "original_status": fci["original_status"],
    "transfer_destination_flag": fci["transfer_destination_flag"],
})

print(fci_processed.shape)
fci_processed.info()

(280, 27)
<class 'pandas.DataFrame'>
RangeIndex: 280 entries, 0 to 279
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype                       
---  ------                     --------------  -----                       
 0   genergy_id                 280 non-null    str                         
 1   oceane_id                  280 non-null    str                         
 2   title                      280 non-null    str                         
 3   description                280 non-null    str                         
 4   application                280 non-null    str                         
 5   status                     280 non-null    str                         
 6   priority                   280 non-null    str                         
 7   category                   280 non-null    str                         
 8   functional_team            280 non-null    str                         
 9   acteur                     279 non-null    s

In [20]:
DATA_PROCESSED_DIR = "../../data/processed/"

fci_processed.to_json(
    DATA_PROCESSED_DIR + "FCI.json", orient="records", date_format="iso", indent=2, force_ascii=False
)
print(f"Wrote {len(fci_processed)} rows to {DATA_PROCESSED_DIR}FCI.json")

Wrote 280 rows to ../../data/processed/FCI.json


`to_json(date_format="iso")` serializes tz-aware timestamps as UTC instants with a `Z` suffix, so
midnight `Europe/Paris` on 2025-10-03 (CEST, UTC+2) shows up as `"2025-10-02T22:00:00.000Z"` — same
instant, different displayed date.

### Remaining flagged rows

Rows with `transfer_destination_flag == "catalogue_not_in_enum"` have no `transferred_to` value:
"Catalogue"/"équipe Catalogue" has no corresponding `TransferDestination` member. Kept in the
exported JSON with `transferred_to = null` rather than dropped or guessed.

In [21]:
fci_processed[fci_processed["transfer_destination_flag"].notna()][
    ["genergy_id", "transfer_destination_flag", "resolution_notes"]
]

,genergy_id,transfer_destination_flag,resolution_notes
9,INC001010856055,catalogue_not_in_enum,14/10: Catalogue: Problème corrigé: La mise à ...
15,INC001010866834,catalogue_not_in_enum,16/10: Catalogue: Bon Usage: Le champ type d'é...
23,INC001010880501,catalogue_not_in_enum,"24/10: Catalogue: Problème non reproduit, le c..."
55,INC001010933003,catalogue_not_in_enum,24_11:SN3: Ticket transféré à l'équipe catalogue.
119,INC001010989963,catalogue_not_in_enum,05_01_SN3: Ticket transféré à l'equipe Catalogue
198,INC001011141577,catalogue_not_in_enum,14_04_SN3 : Ticket trasnféré à l'équipe CATALO...


## COLORIS preprocessing

Same shape as FCI with two differences: the team column is named `Colonne1` instead of
`Paramétrage`, and there are two extra columns, `Offre composée` and `Catégorie d'incident`.

Blank export rows (23 of them) are dropped via null `Statut`, same as FCI.

In [22]:
coloris = coloris.dropna(subset=["Statut"]).reset_index(drop=True)

coloris = coloris.rename(columns={
    "ID Ticket GENERGY ": "genergy_id",
    "ID Ticket OCEANE": "oceane_id",
    "Description": "description",
    "Acteur": "acteur",
    "Colonne1": "team_raw",
    "Statut": "statut_raw",
    "Priorité": "priorite_raw",
    "Actions réalisés": "actions_realisees",
    "Date d'ouverture": "date_ouverture_raw",
    "Date de résolution": "date_resolution_raw",
    "Durée de résolution": "duree_resolution_raw",
    "ID Jira": "jira_id",
    "Offre composée": "offre_composee_raw",
    "Catégorie d'incident": "categorie_incident_raw",
})

print(coloris.shape)
coloris.head()

(286, 14)


,genergy_id,oceane_id,description,acteur,team_raw,statut_raw,priorite_raw,actions_realisees,date_ouverture_raw,date_resolution_raw,duree_resolution_raw,jira_id,offre_composee_raw,categorie_incident_raw
0,INC001010841667,2510K27774,Blocage de FAS sur commandes,ZAKRAOUI Mohamed,Support COLORIS,Résolu,P4,02/10 SN3: Un point planifié avec le demandeur...,02/10/2025,06/10/2025,✅ Résolu en 2 jours,NaN,NaN,NaN
1,INC001010842399,2510K36669,Transfert Coloris-OPUS en rejet,ZAKRAOUI Mohamed,Support COLORIS,Résolu,P4,02/10 SN3:il s’agissait d’un problème lié à OJ...,02/10/2025,02/10/2025,✅ Résolu en < 1 jour,NaN,NaN,NaN
2,INC001010766880,2508V23293,Blocage de resil emplacement,PIERROT CALLIZO Celia,Paramétrage COLORIS,Résolu,P4,En cours de modification pour une mep prevue m...,25/08/2025,06/10/2025,✅ Résolu en 30 jours,NaN,NaN,NaN
3,INC001010785806,2509Z08065,Absence champ type de module dans proxima,PIERROT CALLIZO Celia,Paramétrage COLORIS,Transféré,P4,"Suite analyse, la modification du CI PROXIMA e...",04/09/2025,02/10/2025,➡️ Transféré en 20 jours,NaN,HBNRO,Transféré
4,INC001010835596,2510K20044,6 prestations production alors que commande d...,PIERROT CALLIZO Celia,Paramétrage COLORIS,Résolu,P4,NaN,02/10/2025,09/10/2025,✅ Résolu en 5 jours,NaN,NaN,Paramétrage COLORIS


In [23]:
coloris["jira_id"] = coloris["jira_id"].str.strip()
coloris["requires_jira"] = coloris["jira_id"].notna()

coloris[["jira_id", "requires_jira"]].value_counts(dropna=False)

jira_id                                                                                               requires_jira
NaN                                                                                                   False            266
PCOLORIS-709                                                                                          True               2
PCOLORIS-617                                                                                          True               1
COL-11666                                                                                             True               1
PCOLORIS-623                                                                                          True               1
PCOLORIS-634                                                                                          True               1
PCOLORIS-642                                                                                          True               1
PCOLORIS-644           

In [24]:
# Unlike FCI, a handful of COLORIS rows carry a genuine intraday time (e.g. "07/04/2026 16:26")
# rather than a plain date - format="mixed" preserves it instead of discarding it.
coloris["date_ouverture"] = pd.to_datetime(coloris["date_ouverture_raw"], format="mixed", dayfirst=True)
coloris["date_resolution"] = pd.to_datetime(coloris["date_resolution_raw"], format="mixed", dayfirst=True)

coloris[["date_ouverture_raw", "date_ouverture", "date_resolution_raw", "date_resolution"]].head()

,date_ouverture_raw,date_ouverture,date_resolution_raw,date_resolution
0,02/10/2025,2025-10-02,06/10/2025,2025-10-06
1,02/10/2025,2025-10-02,02/10/2025,2025-10-02
2,25/08/2025,2025-08-25,06/10/2025,2025-10-06
3,04/09/2025,2025-09-04,02/10/2025,2025-10-02
4,02/10/2025,2025-10-02,09/10/2025,2025-10-09


In [25]:
parsed = coloris["duree_resolution_raw"].apply(parse_duree_resolution)
coloris["duree_action"] = parsed.apply(lambda t: t[0])
coloris["duree_jours"] = parsed.apply(lambda t: t[1])

action_to_statut = {"Résolu": "Résolu", "Transféré": "Transféré"}
mismatch_statut = coloris[coloris["duree_action"].map(action_to_statut) != coloris["statut_raw"]]
print("Statut / durée action mismatches:", len(mismatch_statut))

coloris["date_diff_jours"] = (coloris["date_resolution"] - coloris["date_ouverture"]).dt.days
mismatch_days = coloris[coloris["duree_jours"] != coloris["date_diff_jours"]]
print("Duration / date-diff mismatches:", len(mismatch_days))

Statut / durée action mismatches: 0
Duration / date-diff mismatches: 102


In [26]:
assert set(coloris["priorite_raw"].unique()) <= {"P1", "P2", "P3", "P4"}
coloris["priority"] = coloris["priorite_raw"]

STATUT_MAP = {"Résolu": "RESOLVED", "Transféré": "TRANSFERRED"}
assert set(coloris["statut_raw"].unique()) <= set(STATUT_MAP)
coloris["status"] = coloris["statut_raw"].map(STATUT_MAP)

TEAM_MAP = {"Support COLORIS": "SUPPORT", "Paramétrage COLORIS": "CONFIGURATION"}
assert set(coloris["team_raw"].unique()) <= set(TEAM_MAP)
coloris["functional_team"] = coloris["team_raw"].map(TEAM_MAP)

coloris[["priorite_raw", "priority", "statut_raw", "status", "team_raw", "functional_team"]].drop_duplicates()

,priorite_raw,priority,statut_raw,status,team_raw,functional_team
0,P4,P4,Résolu,RESOLVED,Support COLORIS,SUPPORT
2,P4,P4,Résolu,RESOLVED,Paramétrage COLORIS,CONFIGURATION
3,P4,P4,Transféré,TRANSFERRED,Paramétrage COLORIS,CONFIGURATION
36,P4,P4,Transféré,TRANSFERRED,Support COLORIS,SUPPORT
41,P3,P3,Résolu,RESOLVED,Paramétrage COLORIS,CONFIGURATION
60,P3,P3,Résolu,RESOLVED,Support COLORIS,SUPPORT
76,P1,P1,Résolu,RESOLVED,Paramétrage COLORIS,CONFIGURATION
80,P2,P2,Résolu,RESOLVED,Paramétrage COLORIS,CONFIGURATION
122,P3,P3,Transféré,TRANSFERRED,Paramétrage COLORIS,CONFIGURATION


### `title`, `category`, `offer`/`version`

`title` is built the same way as FCI (`truncate_title(description)`). One row
(INC001011250101) has a null `description` with no usable substitute and is dropped rather than
given a fabricated description.

`categorie_incident_raw` (113/309 non-null) is unreliable for most values: 76 rows just repeat the
team name ("Paramétrage COLORIS") and 17 repeat `Statut` ("Transféré"). Only "Bon usage" (20 rows)
is an actual `Category` value, so it's the only value mapped; everything else (including null)
defaults to `CATEGORY_VIDE`.

COLORIS tickets require both `offer` and `version` to be set, but `offre_composee_raw` is sparse
(107/309 rows) and there's no version column. `parse_offer_version` (Shared helpers) extracts both
from that single column: a trailing "V<n>" token becomes `version`, everything before it becomes
`offer`; multiple slash-separated offers keep the first; rows with no `offre_composee_raw` value
get `offer = version = None`. Parsed offer/version values aren't guaranteed to match the current
`Offer`/`Version` enum members — those enums are known to be incomplete and are expected to be
extended separately.

In [27]:
coloris = coloris.dropna(subset=["description"]).reset_index(drop=True)

coloris["title"] = coloris["description"].apply(truncate_title)

coloris["category"] = coloris["categorie_incident_raw"].apply(
    lambda v: "Bon usage" if v == "Bon usage" else "vide"
)

offer_version = coloris["offre_composee_raw"].apply(parse_offer_version)
coloris["offer"] = offer_version.apply(lambda t: t[0])
coloris["version"] = offer_version.apply(lambda t: t[1])

coloris[["description", "title", "categorie_incident_raw", "category", "offre_composee_raw", "offer", "version"]].drop_duplicates(subset=["offre_composee_raw"])

,description,title,categorie_incident_raw,category,offre_composee_raw,offer,version
0,Blocage de FAS sur commandes,Blocage de FAS sur commandes,NaN,vide,NaN,NaN,NaN
3,Absence champ type de module dans proxima,Absence champ type de module dans proxima,Transféré,vide,HBNRO,HBNRO,NaN
55,controle coloris non fonctionnel,controle coloris non fonctionnel,Transféré,vide,DGRPG,DGRPG,NaN
71,Ligne de commande n'est pas modifiable depuis ...,Ligne de commande n'est pas modifiable depuis ...,Paramétrage COLORIS,vide,ROHEB,ROHEB,NaN
72,"Perte du Suffixe sur le champ""Code opération g...","Perte du Suffixe sur le champ""Code opération g...",Paramétrage COLORIS,vide,HEBEP,HEBEP,NaN
73,Redevance travaux emplacement et prestation sp...,Redevance travaux emplacement et prestation sp...,Paramétrage COLORIS,vide,HEBEP/HEPOC,HEBEP,NaN
76,ESIL EMPLACEMENT DOUBLON F01692271025 F9387322...,ESIL EMPLACEMENT DOUBLON F01692271025 F9387322...,Paramétrage COLORIS,vide,DGRPG LOT3,DGRPG LOT3,NaN
79,"Description de la demande : \nBonjour, Le cli...","Description de la demande : \nBonjour, Le clie...",Paramétrage COLORIS,vide,DEPOSE,DEPOSE,NaN
94,"Description de la demande : \nBonjour,J'ai co...","Description de la demande : \nBonjour,J'ai co...",Bon usage,Bon usage,DGRPG V14,DGRPG,V14
98,Description problème : \nLes commandes:\nF3771...,Description problème : \nLes commandes:\nF3771...,Bon usage,Bon usage,ITDMX,ITDMX,NaN


In [28]:
transferred_mask = coloris["status"] == "TRANSFERRED"
results = coloris.loc[transferred_mask, "actions_realisees"].apply(match_transfer_destination)
coloris["transferred_to"] = None
coloris["transfer_destination_flag"] = None
coloris.loc[transferred_mask, "transferred_to"] = results.apply(lambda t: t[0])
coloris.loc[transferred_mask, "transfer_destination_flag"] = results.apply(lambda t: t[1])

print(coloris.loc[transferred_mask, "transferred_to"].value_counts(dropna=False))
print()
print(coloris.loc[transferred_mask, "transfer_destination_flag"].value_counts(dropna=False))

transferred_to
SUPPORT_COLORIS     5
PROXIMA             4
DEVELOPMENT_TEAM    4
SUPPORT_FCI         4
NaN                 2
CONFIG_COLORIS      2
EEP                 1
Name: count, dtype: int64

transfer_destination_flag
NaN                      20
catalogue_not_in_enum     1
no_actions_text           1
Name: count, dtype: int64


In [ ]:
# COLORIS transferred row whose action log contains no destination text.
display(coloris.loc[coloris["transfer_destination_flag"] == "no_actions_text"])

In [29]:
coloris["original_status"] = coloris["status"]
coloris["status"] = "CLOSED"

coloris["created_at"] = coloris["date_ouverture"].dt.tz_localize("Europe/Paris")
coloris["closed_at"] = coloris["date_resolution"].dt.tz_localize("Europe/Paris")
coloris["resolved_at"] = coloris["closed_at"].where(coloris["original_status"] == "RESOLVED")
coloris["updated_at"] = coloris["closed_at"]

coloris["resolution_notes"] = coloris["actions_realisees"]

coloris["transferred_to"] = coloris["transferred_to"].map(DEST_NAME_TO_VALUE)

coloris[[
    "genergy_id", "original_status", "status", "created_at", "resolved_at",
    "closed_at", "transferred_to",
]].head(10)

,genergy_id,original_status,status,created_at,resolved_at,closed_at,transferred_to
0,INC001010841667,RESOLVED,CLOSED,2025-10-02 00:00:00+02:00,2025-10-06 00:00:00+02:00,2025-10-06 00:00:00+02:00,NaN
1,INC001010842399,RESOLVED,CLOSED,2025-10-02 00:00:00+02:00,2025-10-02 00:00:00+02:00,2025-10-02 00:00:00+02:00,NaN
2,INC001010766880,RESOLVED,CLOSED,2025-08-25 00:00:00+02:00,2025-10-06 00:00:00+02:00,2025-10-06 00:00:00+02:00,NaN
3,INC001010785806,TRANSFERRED,CLOSED,2025-09-04 00:00:00+02:00,NaT,2025-10-02 00:00:00+02:00,Proxima
4,INC001010835596,RESOLVED,CLOSED,2025-10-02 00:00:00+02:00,2025-10-09 00:00:00+02:00,2025-10-09 00:00:00+02:00,NaN
5,INC001010842376,RESOLVED,CLOSED,2025-10-03 00:00:00+02:00,2025-10-13 00:00:00+02:00,2025-10-13 00:00:00+02:00,NaN
6,INC001010843878,RESOLVED,CLOSED,2025-10-06 00:00:00+02:00,2025-10-06 00:00:00+02:00,2025-10-06 00:00:00+02:00,NaN
7,INC001010848678,RESOLVED,CLOSED,2025-10-07 00:00:00+02:00,2025-10-08 00:00:00+02:00,2025-10-08 00:00:00+02:00,NaN
8,INC001010847741,RESOLVED,CLOSED,2025-10-06 00:00:00+02:00,2025-10-07 00:00:00+02:00,2025-10-07 00:00:00+02:00,NaN
9,INC001010850508,TRANSFERRED,CLOSED,2025-10-07 00:00:00+02:00,NaT,2025-10-09 00:00:00+02:00,Proxima


## Final assembly and export

Same structure as FCI. `oceane_id`/`jira_delivery_date`/`element`/`vio_app` are `None`
(not applicable to COLORIS); `operational_highlight` defaults to `False`; `archived_at` is `None`.
`offer`/`version` come from the parsed `offre_composee_raw` column instead of being fixed `None`
like FCI.

In [30]:
coloris_processed = pd.DataFrame({
    "genergy_id": coloris["genergy_id"],
    "oceane_id": coloris["oceane_id"],
    "title": coloris["title"],
    "description": coloris["description"],
    "application": "COLORIS",
    "status": coloris["status"],
    "priority": coloris["priority"],
    "category": coloris["category"],
    "functional_team": coloris["functional_team"],
    "acteur": coloris["acteur"],
    "created_at": coloris["created_at"],
    "updated_at": coloris["updated_at"],
    "resolved_at": coloris["resolved_at"],
    "closed_at": coloris["closed_at"],
    "resolution_notes": coloris["resolution_notes"],
    "transferred_to": coloris["transferred_to"],
    "jira_id": coloris["jira_id"],
    "requires_jira": coloris["requires_jira"],
    "jira_delivery_date": None,
    "operational_highlight": False,
    "offer": coloris["offer"],
    "version": coloris["version"],
    "element": None,
    "vio_app": None,
    "archived_at": None,
    "original_status": coloris["original_status"],
    "transfer_destination_flag": coloris["transfer_destination_flag"],
})

print(coloris_processed.shape)
coloris_processed.info()

(285, 27)
<class 'pandas.DataFrame'>
RangeIndex: 285 entries, 0 to 284
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype                       
---  ------                     --------------  -----                       
 0   genergy_id                 281 non-null    str                         
 1   oceane_id                  285 non-null    str                         
 2   title                      285 non-null    str                         
 3   description                285 non-null    str                         
 4   application                285 non-null    str                         
 5   status                     285 non-null    str                         
 6   priority                   285 non-null    str                         
 7   category                   285 non-null    str                         
 8   functional_team            285 non-null    str                         
 9   acteur                     284 non-null    s

In [31]:
coloris_processed.to_json(
    DATA_PROCESSED_DIR + "COLORIS.json", orient="records", date_format="iso", indent=2, force_ascii=False
)
print(f"Wrote {len(coloris_processed)} rows to {DATA_PROCESSED_DIR}COLORIS.json")

Wrote 285 rows to ../../data/processed/COLORIS.json


## AERO preprocessing

Different shape from FCI/COLORIS: no `ID Ticket OCEANE` column, no team/perimeter column, and two
extra columns instead - `ELEMENT` (maps to the `Element` enum, required for AERO tickets) and
`Catégorie` (maps to `Category`, unlike FCI/COLORIS which have no reliable category signal).

`TYPE` is dropped: it's `"AERO"` for all but one row (`"AERO GP"`), redundant with the fixed
`application = "AERO"`.

Blank export rows (139 of them) are dropped via null `Statut`, same convention as FCI/COLORIS.

Every row has `Statut = "Résolu"` - AERO has no `Transféré` rows in this export, so there's no
`transferred_to` matching step for this application.

`ID Ticket GENERGY` is literally `"mail"` on 6 rows (reported by email rather than through the
ticket system) - kept as-is rather than nulled out, per prior decision.

In [32]:
aero = aero.dropna(subset=["Statut"]).reset_index(drop=True)

aero = aero.rename(columns={
    "ID Ticket GENERGY ": "genergy_id",
    "Description": "description",
    "Acteur": "acteur",
    "Catégorie ": "categorie_raw",
    "Statut": "statut_raw",
    "Priorité": "priorite_raw",
    "Actions réalisés": "actions_realisees",
    "Date d'ouverture": "date_ouverture_raw",
    "Date de résolution": "date_resolution_raw",
    "Durée de résolution": "duree_resolution_jours",
    "ID Jira": "jira_id",
    "ELEMENT": "element_raw",
})
aero = aero.drop(columns=["TYPE"])

print(aero.shape)
aero.head()

(54, 12)


,genergy_id,element_raw,description,acteur,categorie_raw,statut_raw,priorite_raw,actions_realisees,date_ouverture_raw,date_resolution_raw,duree_resolution_jours,jira_id
0,mail,FIBRE IHM-FTTE,Pb dysfonctionnement Eligibilité FTTE sur RIP ...,Akram Sahli,Demande d'assistance,Résolu,P3,"Après vérification, les résultats d’éligibilit...",01/12/2025 00:00,02/12/2025 00:00,2.0,NaN
1,INC001010977278,FIBRE IHM-FTTH,Il est indiqué que le code immeuble IMB/33119/...,Mariem Hammami,Demande d'assistance,Résolu,P3,l’IMB n’est pas présente dans les fichiers del...,19/12/2025,19/12/2025 00:00,1.0,NaN
2,INC001010986955,Orange-Eligibility KO,WS Orange KO deouis 14:50,Akram Sahli,Bug,Résolu,P3,Le problème a été résolu hier à 17h40.\nToutef...,29/12/2025,29/12/2025,1.0,NaN
3,INC001011008164,DONNEES ABSENTES OU INCORRECTES,Nous constatons une erreur sur le PM lorsqu'on...,Akram Sahli,Hors perimètre d'AERO,Résolu,P3,le PM OAPC n'est pas dans notre périmètre d'in...,14/01/2026,14/01/2026,1.0,NaN
4,INC001011016378,CONNEXION IMPOSSIBLE,Depuis 19/01/2026 08h00 on a erreurs 400 sur P...,Akram Sahli,Aucun probléme lié a AERO,Résolu,P3,"Après analyse des logs, les requêtes ne semble...",20/01/2026,21/01/2026 00:00,2.0,NaN


In [33]:
aero["jira_id"] = aero["jira_id"].str.strip()
aero["requires_jira"] = aero["jira_id"].notna()

aero[["jira_id", "requires_jira"]].value_counts(dropna=False)

jira_id    requires_jira
NaN        False            41
AERO-5147  True              3
AERO-5118  True              2
AERO-5192  True              2
AERO-5279  True              2
AERO-5130  True              1
AERO-5153  True              1
AERO-5195  True              1
AERO-5285  True              1
Name: count, dtype: int64

In [34]:
aero["date_ouverture"] = pd.to_datetime(aero["date_ouverture_raw"], format="mixed", dayfirst=True)
aero["date_resolution"] = pd.to_datetime(aero["date_resolution_raw"], format="mixed", dayfirst=True)

# duree_resolution_jours is already a plain day count here (not the emoji-prefixed FCI/COLORIS
# string) - cross-check directly against the calendar day difference.
aero["date_diff_jours"] = (aero["date_resolution"] - aero["date_ouverture"]).dt.days
mismatch_days = aero[aero["duree_resolution_jours"] != aero["date_diff_jours"]]
print("Duration / date-diff mismatches:", len(mismatch_days))

aero[["date_ouverture_raw", "date_ouverture", "date_resolution_raw", "date_resolution", "duree_resolution_jours", "date_diff_jours"]].head()

Duration / date-diff mismatches: 54


,date_ouverture_raw,date_ouverture,date_resolution_raw,date_resolution,duree_resolution_jours,date_diff_jours
0,01/12/2025 00:00,2025-12-01,02/12/2025 00:00,2025-12-02,2.0,1
1,19/12/2025,2025-12-19,19/12/2025 00:00,2025-12-19,1.0,0
2,29/12/2025,2025-12-29,29/12/2025,2025-12-29,1.0,0
3,14/01/2026,2026-01-14,14/01/2026,2026-01-14,1.0,0
4,20/01/2026,2026-01-20,21/01/2026 00:00,2026-01-21,2.0,1


In [35]:
assert set(aero["priorite_raw"].unique()) <= {"P1", "P2", "P3", "P4"}
aero["priority"] = aero["priorite_raw"]

assert set(aero["statut_raw"].unique()) == {"Résolu"}
aero["status"] = "RESOLVED"

# No team/perimeter column in this export - fixed to SUPPORT for every AERO row.
aero["functional_team"] = "SUPPORT"

aero[["priorite_raw", "priority", "statut_raw", "status", "functional_team"]].drop_duplicates()

,priorite_raw,priority,statut_raw,status,functional_team
0,P3,P3,Résolu,RESOLVED,SUPPORT
14,P4,P4,Résolu,RESOLVED,SUPPORT
19,P2,P2,Résolu,RESOLVED,SUPPORT
45,P1,P1,Résolu,RESOLVED,SUPPORT


### `title`, `element`, `category`

`title` is built the same way as FCI/COLORIS.

`element_raw` matches the `Element` enum values after normalizing case/accents/dashes, except one
typo ("INATENDU" instead of "INATTENDU") corrected explicitly.

`categorie_raw` maps to `Category` with best-effort semantic matches plus two explicit overrides:
"Aucun problème lié à AERO" → `CATEGORY_HORS_PERIMETRE` (same as "Hors périmètre d'AERO") and
"Infrastructre" → `CATEGORY_INFRASTRUCTURE` (enum member added for this).

In [36]:
from app.modules.ticket_management.domain.enums.element import Element


def _normalize_element(text: str) -> str:
    return normalize_text(text).replace("-", " ").strip()


ELEMENT_BY_NORM = {_normalize_element(e.value): e.name for e in Element}
# Source typo: "INATENDU" instead of "INATTENDU".
ELEMENT_BY_NORM[_normalize_element("MESSAGE D'ERREUR OU RESULTAT INATENDU")] = "ERREUR"

aero["element"] = aero["element_raw"].apply(
    lambda v: Element[ELEMENT_BY_NORM[_normalize_element(v)]].value
)

CATEGORIE_MAP = {
    "Bug": "Bug",
    "Demande d'assistance": "Assistance client",
    "Hors perimètre d'AERO": "Hors périmètre",
    "Information Manquante": "Manque d'information",
    "Aucun probléme lié a AERO ": "Hors périmètre",
    "Infrastructre": "Infrastructure",
}
assert set(aero["categorie_raw"].unique()) <= set(CATEGORIE_MAP)
aero["category"] = aero["categorie_raw"].map(CATEGORIE_MAP)

aero["title"] = aero["description"].apply(truncate_title)

aero[["element_raw", "element", "categorie_raw", "category"]].drop_duplicates()

,element_raw,element,categorie_raw,category
0,FIBRE IHM-FTTE,Fibre IHM FTTE,Demande d'assistance,Assistance client
1,FIBRE IHM-FTTH,Fibre IHM FTTH,Demande d'assistance,Assistance client
2,Orange-Eligibility KO,Orange-Eligibility KO,Bug,Bug
3,DONNEES ABSENTES OU INCORRECTES,Données absentes ou incorrectes,Hors perimètre d'AERO,Hors périmètre
4,CONNEXION IMPOSSIBLE,Connexion impossible,Aucun probléme lié a AERO,Hors périmètre
8,Orange-Eligibility KO,Orange-Eligibility KO,Demande d'assistance,Assistance client
9,API,API,Demande d'assistance,Assistance client
11,CUIVRE IHM,Cuivre IHM,Demande d'assistance,Assistance client
12,FIBRE IHM-FTTH,Fibre IHM FTTH,Bug,Bug
13,RETOUR SUR TICKET,Retour sur ticket,Bug,Bug


In [37]:
aero["original_status"] = aero["status"]
aero["status"] = "CLOSED"

aero["created_at"] = aero["date_ouverture"].dt.tz_localize("Europe/Paris")
aero["closed_at"] = aero["date_resolution"].dt.tz_localize("Europe/Paris")
aero["resolved_at"] = aero["closed_at"]
aero["updated_at"] = aero["closed_at"]

aero["resolution_notes"] = aero["actions_realisees"]
aero["transferred_to"] = None
aero["transfer_destination_flag"] = None

aero[["genergy_id", "original_status", "status", "created_at", "resolved_at", "closed_at"]].head(10)

,genergy_id,original_status,status,created_at,resolved_at,closed_at
0,mail,RESOLVED,CLOSED,2025-12-01 00:00:00+01:00,2025-12-02 00:00:00+01:00,2025-12-02 00:00:00+01:00
1,INC001010977278,RESOLVED,CLOSED,2025-12-19 00:00:00+01:00,2025-12-19 00:00:00+01:00,2025-12-19 00:00:00+01:00
2,INC001010986955,RESOLVED,CLOSED,2025-12-29 00:00:00+01:00,2025-12-29 00:00:00+01:00,2025-12-29 00:00:00+01:00
3,INC001011008164,RESOLVED,CLOSED,2026-01-14 00:00:00+01:00,2026-01-14 00:00:00+01:00,2026-01-14 00:00:00+01:00
4,INC001011016378,RESOLVED,CLOSED,2026-01-20 00:00:00+01:00,2026-01-21 00:00:00+01:00,2026-01-21 00:00:00+01:00
5,INC001011016287,RESOLVED,CLOSED,2026-01-20 00:00:00+01:00,2026-01-21 00:00:00+01:00,2026-01-21 00:00:00+01:00
6,INC001011022462,RESOLVED,CLOSED,2026-01-22 00:00:00+01:00,2026-01-22 00:00:00+01:00,2026-01-22 00:00:00+01:00
7,INC001011028215,RESOLVED,CLOSED,2026-01-27 00:00:00+01:00,2026-01-27 00:00:00+01:00,2026-01-27 00:00:00+01:00
8,INC001011035776,RESOLVED,CLOSED,2026-01-30 00:00:00+01:00,2026-02-02 00:00:00+01:00,2026-02-02 00:00:00+01:00
9,mail,RESOLVED,CLOSED,2026-02-03 00:00:00+01:00,2026-02-03 00:00:00+01:00,2026-02-03 00:00:00+01:00


## Final assembly and export

Same structure as the other applications. `oceane_id` is `None` (no such column in this export);
`offer`/`version`/`vio_app` are `None` (not applicable to AERO); `jira_delivery_date` is `None`;
`operational_highlight` defaults to `False`; `archived_at` is `None`.

In [38]:
aero_processed = pd.DataFrame({
    "genergy_id": aero["genergy_id"],
    "oceane_id": None,
    "title": aero["title"],
    "description": aero["description"],
    "application": "AERO",
    "status": aero["status"],
    "priority": aero["priority"],
    "category": aero["category"],
    "functional_team": aero["functional_team"],
    "acteur": aero["acteur"],
    "created_at": aero["created_at"],
    "updated_at": aero["updated_at"],
    "resolved_at": aero["resolved_at"],
    "closed_at": aero["closed_at"],
    "resolution_notes": aero["resolution_notes"],
    "transferred_to": aero["transferred_to"],
    "jira_id": aero["jira_id"],
    "requires_jira": aero["requires_jira"],
    "jira_delivery_date": None,
    "operational_highlight": False,
    "offer": None,
    "version": None,
    "element": aero["element"],
    "vio_app": None,
    "archived_at": None,
    "original_status": aero["original_status"],
    "transfer_destination_flag": aero["transfer_destination_flag"],
})

print(aero_processed.shape)
aero_processed.info()

(54, 27)
<class 'pandas.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype                       
---  ------                     --------------  -----                       
 0   genergy_id                 54 non-null     str                         
 1   oceane_id                  0 non-null      object                      
 2   title                      54 non-null     str                         
 3   description                54 non-null     str                         
 4   application                54 non-null     str                         
 5   status                     54 non-null     str                         
 6   priority                   54 non-null     str                         
 7   category                   54 non-null     str                         
 8   functional_team            54 non-null     str                         
 9   acteur                     53 non-null     str 

In [39]:
aero_processed.to_json(
    DATA_PROCESSED_DIR + "AERO.json", orient="records", date_format="iso", indent=2, force_ascii=False
)
print(f"Wrote {len(aero_processed)} rows to {DATA_PROCESSED_DIR}AERO.json")

Wrote 54 rows to ../../data/processed/AERO.json


## VIO preprocessing

Simpler shape than FCI/COLORIS: `App` maps to the `VioApp` enum (required for VIO tickets),
`Périmètre` is constantly `"Support VIO"` (→ `functional_team = SUPPORT` for every row), `Statut`
is constantly `"Résolu"` (no transfers, so no `transferred_to` matching step), and there's no
category column at all (defaults to `CATEGORY_VIDE`, same as FCI).

Blank export rows (10 of them) are dropped via null `Statut`, same convention as the other files.

In [40]:
vio = vio.dropna(subset=["Statut"]).reset_index(drop=True)

vio = vio.rename(columns={
    "ID Ticket GENERGY ": "genergy_id",
    "ID Ticket OCEANE": "oceane_id",
    "Description": "description",
    "Acteur": "acteur",
    "App": "vio_app_raw",
    "Périmètre ": "perimetre_raw",
    "Statut": "statut_raw",
    "Priorité": "priorite_raw",
    "Actions réalisés": "actions_realisees",
    "Date d'ouverture": "date_ouverture_raw",
    "Date de résolution": "date_resolution_raw",
    "Durée de résolution": "duree_resolution_raw",
    "ID Jira": "jira_id",
})

print(vio.shape)
vio.head()

(180, 13)


,genergy_id,oceane_id,vio_app_raw,description,acteur,perimetre_raw,statut_raw,priorite_raw,actions_realisees,date_ouverture_raw,date_resolution_raw,duree_resolution_raw,jira_id
0,INC001010961694,2512K27406,Parc,Problème Marine,Mariem Hammami,Support VIO,Résolu,P3,L’investigation a été effectuée,10/11/2025,11/12/2025 00:00,✅ Résolu en 23 jours,NaN
1,INC001010946366,2512H51825,Parc,Dans le cadre d'une mise à jour du SI et du tr...,Akram Sahli,Support VIO,Résolu,P4,Nous avons bien pris en compte votre demande c...,02/12/2025,04/12/2025,✅ Résolu en 2 jours,NaN
2,INC001010954206,2512J75412,Parc,J'ai besoin du parc d'accès du client CEGETEL ...,Akram Sahli,Support VIO,Résolu,P3,Nous avons bien pris en compte votre demande c...,09/12/2025,10/12/2025,✅ Résolu en 1 jour,NaN
3,NaN,2512L11587,Vigie,"[VIGIE_VI] / [SMC4, BRONZE, SOX] / [REMCOLL_NO...",Akram Sahli,Support VIO,Résolu,P4,"Bonjour,\nL’incident a été rétabli le lendemai...",12/12/2025,18/12/2025 00:00,✅ Résolu en 4 jours,NaN
4,NaN,2512L01771,Sagic,"[SAGIC] / [SMC3, ARGENT] / [OBV_APP_FILE_STRIN...",Mariem Hammami,Support VIO,Résolu,P3,Voir avec la MOE si le problème est récurrent....,12/12/2025,19/01/1900,✅ Résolu en < 1 jour,NaN


In [41]:
vio["jira_id"] = vio["jira_id"].str.strip()
vio["requires_jira"] = vio["jira_id"].notna()

vio[["jira_id", "requires_jira"]].value_counts(dropna=False)

jira_id   requires_jira
NaN       False            174
VIO-4323  True               1
VIO-4389  True               1
VIO-4410  True               1
VIO-4539  True               1
VIO-4605  True               1
VIO-4647  True               1
Name: count, dtype: int64

One row (`ID Ticket OCEANE` `2512L01771`) has `Date de résolution = "19/01/1900"`, an obviously
bogus placeholder (opened 2025-12-12, duration text says "< 1 jour"). Per prior decision, its
resolution date is substituted with its opening date rather than kept as-is or left null.

In [42]:
bogus_resolution_date = vio["date_resolution_raw"] == "19/01/1900"
vio.loc[bogus_resolution_date, "date_resolution_raw"] = vio.loc[bogus_resolution_date, "date_ouverture_raw"]

vio["date_ouverture"] = pd.to_datetime(vio["date_ouverture_raw"], format="mixed", dayfirst=True)
vio["date_resolution"] = pd.to_datetime(vio["date_resolution_raw"], format="mixed", dayfirst=True)

parsed = vio["duree_resolution_raw"].apply(parse_duree_resolution)
vio["duree_jours"] = parsed.apply(lambda t: t[1])
vio["date_diff_jours"] = (vio["date_resolution"] - vio["date_ouverture"]).dt.days
mismatch_days = vio[vio["duree_jours"] != vio["date_diff_jours"]]
print("Duration / date-diff mismatches:", len(mismatch_days))

vio[["date_ouverture_raw", "date_ouverture", "date_resolution_raw", "date_resolution"]].head()

Duration / date-diff mismatches: 39


,date_ouverture_raw,date_ouverture,date_resolution_raw,date_resolution
0,10/11/2025,2025-11-10,11/12/2025 00:00,2025-12-11
1,02/12/2025,2025-12-02,04/12/2025,2025-12-04
2,09/12/2025,2025-12-09,10/12/2025,2025-12-10
3,12/12/2025,2025-12-12,18/12/2025 00:00,2025-12-18
4,12/12/2025,2025-12-12,12/12/2025,2025-12-12


In [43]:
from app.modules.ticket_management.domain.enums.vio_app import VioApp

assert set(vio["priorite_raw"].unique()) <= {"P1", "P2", "P3", "P4"}
vio["priority"] = vio["priorite_raw"]

assert set(vio["statut_raw"].unique()) == {"Résolu"}
vio["status"] = "RESOLVED"

assert set(vio["perimetre_raw"].unique()) == {"Support VIO"}
vio["functional_team"] = "SUPPORT"

VIO_APP_MAP = {"Parc": "PARC", "Vigie": "VIGIE", "Sagic": "SAGIC", "Fop ": "FOP"}
assert set(vio["vio_app_raw"].unique()) <= set(VIO_APP_MAP)
vio["vio_app"] = vio["vio_app_raw"].map(VIO_APP_MAP).map(lambda name: VioApp[name].value)

vio["title"] = vio["description"].apply(truncate_title)
vio["category"] = "vide"

vio[["vio_app_raw", "vio_app", "priority", "status", "functional_team", "category"]].drop_duplicates()

,vio_app_raw,vio_app,priority,status,functional_team,category
0,Parc,PARC,P3,RESOLVED,SUPPORT,vide
1,Parc,PARC,P4,RESOLVED,SUPPORT,vide
3,Vigie,VIGIE,P4,RESOLVED,SUPPORT,vide
4,Sagic,SAGIC,P3,RESOLVED,SUPPORT,vide
6,Fop,FOP,P2,RESOLVED,SUPPORT,vide
7,Fop,FOP,P3,RESOLVED,SUPPORT,vide
26,Sagic,SAGIC,P2,RESOLVED,SUPPORT,vide
36,Sagic,SAGIC,P4,RESOLVED,SUPPORT,vide
37,Fop,FOP,P4,RESOLVED,SUPPORT,vide
44,Parc,PARC,P2,RESOLVED,SUPPORT,vide


In [44]:
vio["original_status"] = vio["status"]
vio["status"] = "CLOSED"

vio["created_at"] = vio["date_ouverture"].dt.tz_localize("Europe/Paris")
vio["closed_at"] = vio["date_resolution"].dt.tz_localize("Europe/Paris")
vio["resolved_at"] = vio["closed_at"]
vio["updated_at"] = vio["closed_at"]

vio["resolution_notes"] = vio["actions_realisees"]
vio["transferred_to"] = None
vio["transfer_destination_flag"] = None

vio[["oceane_id", "original_status", "status", "created_at", "resolved_at", "closed_at", "vio_app"]].head(10)

,oceane_id,original_status,status,created_at,resolved_at,closed_at,vio_app
0,2512K27406,RESOLVED,CLOSED,2025-11-10 00:00:00+01:00,2025-12-11 00:00:00+01:00,2025-12-11 00:00:00+01:00,PARC
1,2512H51825,RESOLVED,CLOSED,2025-12-02 00:00:00+01:00,2025-12-04 00:00:00+01:00,2025-12-04 00:00:00+01:00,PARC
2,2512J75412,RESOLVED,CLOSED,2025-12-09 00:00:00+01:00,2025-12-10 00:00:00+01:00,2025-12-10 00:00:00+01:00,PARC
3,2512L11587,RESOLVED,CLOSED,2025-12-12 00:00:00+01:00,2025-12-18 00:00:00+01:00,2025-12-18 00:00:00+01:00,VIGIE
4,2512L01771,RESOLVED,CLOSED,2025-12-12 00:00:00+01:00,2025-12-12 00:00:00+01:00,2025-12-12 00:00:00+01:00,SAGIC
5,2512K66016,RESOLVED,CLOSED,2025-12-11 00:00:00+01:00,2025-12-18 00:00:00+01:00,2025-12-18 00:00:00+01:00,PARC
6,2512N01311,RESOLVED,CLOSED,2025-12-19 00:00:00+01:00,2025-12-19 00:00:00+01:00,2025-12-19 00:00:00+01:00,FOP
7,2512P58372,RESOLVED,CLOSED,2025-12-26 00:00:00+01:00,2025-12-29 00:00:00+01:00,2025-12-29 00:00:00+01:00,FOP
8,2512O91480,RESOLVED,CLOSED,2025-12-26 00:00:00+01:00,2025-12-29 00:00:00+01:00,2025-12-29 00:00:00+01:00,FOP
9,2512O96520,RESOLVED,CLOSED,2025-12-26 00:00:00+01:00,2025-12-29 00:00:00+01:00,2025-12-29 00:00:00+01:00,FOP


## Final assembly and export

Same structure as the other applications. `offer`/`version`/`element` are `None` (not applicable
to VIO); `jira_delivery_date` is `None`; `operational_highlight` defaults to `False`;
`archived_at` is `None`.

In [45]:
vio_processed = pd.DataFrame({
    "genergy_id": vio["genergy_id"],
    "oceane_id": vio["oceane_id"],
    "title": vio["title"],
    "description": vio["description"],
    "application": "VIO",
    "status": vio["status"],
    "priority": vio["priority"],
    "category": vio["category"],
    "functional_team": vio["functional_team"],
    "acteur": vio["acteur"],
    "created_at": vio["created_at"],
    "updated_at": vio["updated_at"],
    "resolved_at": vio["resolved_at"],
    "closed_at": vio["closed_at"],
    "resolution_notes": vio["resolution_notes"],
    "transferred_to": vio["transferred_to"],
    "jira_id": vio["jira_id"],
    "requires_jira": vio["requires_jira"],
    "jira_delivery_date": None,
    "operational_highlight": False,
    "offer": None,
    "version": None,
    "element": None,
    "vio_app": vio["vio_app"],
    "archived_at": None,
    "original_status": vio["original_status"],
    "transfer_destination_flag": vio["transfer_destination_flag"],
})

print(vio_processed.shape)
vio_processed.info()

(180, 27)
<class 'pandas.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype                       
---  ------                     --------------  -----                       
 0   genergy_id                 4 non-null      str                         
 1   oceane_id                  180 non-null    str                         
 2   title                      180 non-null    str                         
 3   description                180 non-null    str                         
 4   application                180 non-null    str                         
 5   status                     180 non-null    str                         
 6   priority                   180 non-null    str                         
 7   category                   180 non-null    str                         
 8   functional_team            180 non-null    str                         
 9   acteur                     180 non-null    s

In [46]:
vio_processed.to_json(
    DATA_PROCESSED_DIR + "VIO.json", orient="records", date_format="iso", indent=2, force_ascii=False
)
print(f"Wrote {len(vio_processed)} rows to {DATA_PROCESSED_DIR}VIO.json")

Wrote 180 rows to ../../data/processed/VIO.json


## Conclusion and flagged notes:

- ``FCI``: 280 rows, 6 flagged (catalogue_not_in_enum, no matching enum value)
- ``COLORIS``: 285 rows, 2 flagged (1 catalogue, 1 no destination text) — offer/version parsed from the composite offer column
- ``AERO``: 54 rows, element/category mapped from dedicated columns, 6 "mail"-origin rows preserved as-is
- ``VIO``: 180 rows, vio_app mapped, one bogus 1900 date corrected